src/PotentialCalculation/SuccessiveOverRelaxation/SOR.jl

```diff
@@ -71,33 +72,55 @@ Thus, there cannot be local extrema.
 """
 @inline @fastmath function handle_depletion(
     new_potential::T, 
-    point_type::PointType, 
+    imp_scale::T,
     neighbor_potentials::NTuple{6,T}, 
     q_eff_imp::T, 
     volume_weight::T,
     sor_const::T
-)::Tuple{T, PointType} where {T}
+)::Tuple{T, T} where {T}
     vmin::T = min(neighbor_potentials...)
     vmax::T = max(neighbor_potentials...)
-
-    if new_potential < vmin || new_potential > vmax
-        new_potential -= q_eff_imp * volume_weight * sor_const
-        if (point_type & undepleted_bit == 0 && point_type & pn_junction_bit > 0) 
-            point_type += undepleted_bit
+    
+    is_ptype = q_eff_imp < 0 
+    neighbor_relevant_extremum = is_ptype ? vmin : vmax
+    imp_contribution = q_eff_imp * volume_weight * sor_const 
+    
+    if !iszero(imp_contribution) 
+        new_potential_proposal = new_potential - (T(1)- imp_scale) * imp_contribution
+        if (new_potential_proposal > vmax || new_potential_proposal < vmin)
+            imp_scale::T = (imp_contribution - (new_potential - neighbor_relevant_extremum)) / (imp_contribution * sor_const)
+            if imp_scale < 0 imp_scale = T(0) end
+            if imp_scale > 1 imp_scale = T(1) end
```

Comment:
This is exactly what `min` and `max` do:
```julia
imp_scale = max(imp_scale, T(0))
imp_scale = min(imp_scale, T(1))
```
or
```julia
imp_scale = min(max(imp_scale, T(0)), T(1))
```

There might even be a function where you give the edges of an interval and it does exactly this.

Reply:
This can be (is) done.